src/interfaces/README.md

```diff
@@ -12,6 +12,8 @@ The following interfaces are defined here:
 
 * [`Handler`](handler.h) — returned by `handleEvent` methods on interfaces above and used to manage lifetimes of event handlers.
 
-* [`Init`](init.h) — used by multiprocess code to access interfaces above on startup. Added in [#10102](https://github.com/bitcoin/bitcoin/pull/10102).
+* [`Init`](init.h) — used by multiprocess code to access interfaces above on startup. Added in [#19160](https://github.com/bitcoin/bitcoin/pull/19160).
+
+* [`Base`](base.h) — base interface class used by multiprocess code for bookkeeping and cleanup. Added in [#19160](https://github.com/bitcoin/bitcoin/pull/19160).
 
```

Comment:
It would be great to extend the README with a new section explaining how someone would add a new `IpcProtocol` implementation. As of right now `src/interfaces/capnp/init.capnp` is pretty empty which makes it hard to understand for reviewers without looking first on #10102 branch.

Purely reading libmultiprocess README, I didn't grasp at first than one `IpcProtocol` must define a `ProxyClient`/ProxyServer` pair for each supported interface.

Reply:
> I'm not sure what change is suggested by the observation about init.capnp. I hope it's clear that things in the src/interfaces/capnp/ directory are specific to the capnp protocol.

I think it's clearer for me now about the distinction between abstract interfaces and their implementations. IMO, three further improvements to make it better:
* rename `IpcProtocolImpl` to `CapnpProtocol`, the constructor is already called `MakeCapnpProtocol` and `IpcProcessImpl` to `LibmpProcess`. Or any more accurate names hinting the _specific_ implementation
* document `IpcProtocol`/`IpcProcess` in `src/interfaces/README.md`, underscoring the current available implementation `IpcProtocolImpl`/`IpcProcessImpl` for each of them
* reorganize directory to `src/interfaces/capnp/` `src/interfaces/ipcproto/capnp` underscoring that capnp is only one implementation among potential several ones.

> It's not the case that an IPC protocol needs to be implemented with pairs of client and server classes. 

I did looked again on the #10102 and this point is more straightforward to understand. That said, for this current PR, I believe it will helper reviewers to make a reference on how pairs of clients and server are added for capnp implementation. 

I don't know if a `src/interfaces/capnp/README.md` with implementation details would help but for example I still don't grasp how `.capnp` files are consumed by the code generator. I might need to read `libmultiprocess/README.md` more attentively though.